rpc/src/client.rs

```diff
@@ -180,3 +180,188 @@ impl Client {
         }
     }
 }
+
+impl Client<ClientConnectionAsync> {
```

Comment:
I think it's better to use two different struct name rather than use template. Here are two reasons:
1. user do not aware about `ClientConnection`.
2. we will take off the thread version in the future. The code using async version will not change.

So, maybe you can just implement `Client` with async and rename the old `Client` as `ClientSync`.

Reply:
`ClientSync` is an example, you can choose any other name.

Reply:
done, Changed to `ClientSync`